sql/group.go

```diff
@@ -125,6 +130,19 @@ func (n *groupNode) Next() bool {
 	for n.plan.Next() {
 		values := n.plan.Values()
 		for i, f := range n.funcs {
+			if f.val.Distinct {
```

Comment:
If you move the `seen` map inside of `aggregateFunc`, I think this body of code should move to `aggregateFunc.Add` which would perform the distinct processing before calling through to `aggregateFunc.impl`. Oh, and this test should be something like `n.seen[i] != nil` rather than examining `f.val.Distinct` again.